### PR TITLE
DSERV-601-load-SEM-preds

### PR DESCRIPTION
**Pull request by shengchd**
Adapters for loading motifs & variants_proteins predictions from SEM

### CODE REVIEW

File: data/adapters/SEM_motif_adapter.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                    tf_id = self.tf_id_mapping.get(tf_name)
+                    if tf_id is None:
+                        print(
+                            'TF id unavailable, skipping motif_protein_link: ' + tf_name)
+                        continue
+
+                    _key = motif_id + '_' + SEMMotif.SOURCE + \
+                        '_' + tf_id.split('/')[-1]
+                    _from = 'motifs/' + tf_name + '_' + SEMMotif.SOURCE
+                    _to = tf_id  # nodes in either proteins or complexes -> should the schema be updated?
```
Comment (shengchd) [inline data/adapters/SEM_motif_adapter.py:108]: not sure how to/if needed to represent the mixture case in schema

File: data/adapters/SEM_motif_adapter.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                        'name': _key,
+                        'tf_name': tf_name,
+                        'source': SEMMotif.SOURCE,
+                        'source_url': SEMMotif.SOURCE_URL,
+                        'pwm': pwm,
+                        'length': length
+                    }
+                elif self.label == 'motif_protein_link':
+                    self.load_tf_id_mapping()
+                    with open(self.filepath + '/' + filename, 'r') as sem_file:
```
Comment (mingjiecn) [inline data/adapters/SEM_motif_adapter.py:96]: Those two lines are duplicated, can you refactor?

**Review by mingjiecn — APPROVED** [review]
Code looks good. Can you refactor the duplication?

**Review by mingjiecn — CHANGES_REQUESTED** [review]
I think the loading will have problem if the collection is actually mixed. Did you tried load some sample data?

**Comment by shengchd** [comment]
> I think the loading will have problem if the collection is actually mixed. Did you tried load some sample data?

File: data/adapters/SEM_motif_adapter.py (hunk 2 above)
Comment (shengchd) [reply data/adapters/SEM_motif_adapter.py]: yes refactored.

**Comment by shengchd** [comment]
> > I think the loading will have problem if the collection is actually mixed. Did you tried load some sample data?

tried loading test data in motifs_proteins_test, seems to work fine.

**Review by mingjiecn — APPROVED** [review]

**Review by mingjiecn — APPROVED** [review]

File: data/adapters/SEM_prediction_adapter.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                            _to = tf_id
+
+                            _props = {
+                                '_key': _key,
+                                '_from': _from,
+                                '_to': _to,
+                                'label': 'predicted allele specific binding',
+                                'motif': 'motifs/' + tf_name + '_' + SEMPred.SOURCE,
+                                'kmer_chr': row[6].split(':')[0],
+                                'kmer_start': int(row[6].split(':')[-1].split('-')[0]),
```
Comment (mingjiecn) [inline data/adapters/SEM_prediction_adapter.py:90]: I think `kmer_start` need to be updated to `kmer_start:long`. Same with `kmer_end`

**Pull request merged**